operator/src/main/java/org/bf2/operator/operands/AbstractKafkaCluster.java

```diff
@@ -417,17 +416,18 @@ protected List<GenericKafkaListener> buildListeners(ManagedKafka managedKafka, i
                                 .withAuth(oauthAuthenticationListener)
                                 .withNetworkPolicyPeers(new NetworkPolicyPeerBuilder()
                                         .withNewPodSelector()
-                                        .addToMatchLabels("app", AbstractAdminServer.adminServerName(managedKafka))
-                                        .endPodSelector().build())
+                                            .addToMatchLabels("app", AbstractAdminServer.adminServerName(managedKafka))
+                                        .endPodSelector()
+                                        .build())
                                 .build(),
                         new GenericKafkaListenerBuilder()
                                 .withName("sre")
                                 .withPort(9096)
                                 .withType(KafkaListenerType.INTERNAL)
                                 .withNetworkPolicyPeers(new NetworkPolicyPeerBuilder()
-                                        .withIpBlock(new IPBlockBuilder()
-                                                .withCidr("0.0.0.0/32")
-                                                .build())
+                                        .withNewPodSelector()
+                                            .addToMatchLabels("strimzi.io/name", managedKafka.getMetadata().getName() + "-kafka")
```

Comment:
Don't we need a namespace selector too? Otherwise an evil person could label any old pod with that label and have access?

https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-metadata-name

Reply:
I thought that as well, but I think this applies [1]:

> podSelector: This selects particular Pods in the same namespace as the NetworkPolicy which should be allowed as ingress sources or egress destinations.

[1] https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors

Reply:
Nonetheless. this thread is making me question it so I'll try that :-)